newsroom/template_filters.py

```diff
@@ -61,6 +70,113 @@ def parse_date(datetime):
+        return ScheduleType.MULTI_DAY
+
+    if duration == DAY_IN_MINUTES and start.date() == (end.date() if end else None):
+        return ScheduleType.ALL_DAY
+
+    if duration == 0 and start.time() == (end.time() if end else None):
+        return ScheduleType.NO_DURATION
+
+    return ScheduleType.REGULAR
+
+
+def is_item_tbc(item: dict) -> bool:
+    event_tbc = item.get("event", {}).get("_time_to_be_confirmed", False)
+    planning = item.get("planning_items", [])
+    return event_tbc or (planning and planning[0].get("_time_to_be_confirmed", False))
+
+
+def format_event_datetime(item: dict) -> str:
+    date_info = item.get("dates", {})
+
+    if not date_info:
+        return ""
+
+    tz = date_info.get("tz", get_session_timezone())
+    try:
+        # Set the session timezone
+        set_session_timezone(tz)
+
+        start = parse_date(format_datetime(parse_date(date_info.get("start")), "yyyy-MM-dd HH:mm:ss"))
+        end = (
+            parse_date(format_datetime(parse_date(date_info.get("end")), "yyyy-MM-dd HH:mm:ss"))
+            if date_info.get("end")
+            else None
+        )
+        all_day = date_info.get("all_day")
+        no_end_time = date_info.get("no_end_time")
+        is_tbc_item = is_item_tbc(item)
+
+        schedule_type = get_schedule_type(start, end, all_day, no_end_time)
+
+        formatted_start = datetime_long(start)
+        formatted_end = datetime_long(end) if end else None
+
+        if is_tbc_item:
+            if start.date() != (end.date() if end else None):
+                return lazy_gettext("Event Starts: {start} - Event Ends: {end} ({tz}) (Time to be confirmed)").format(
+                    start=formatted_start, end=formatted_end, tz=tz
+                )
+
+            else:
+                return lazy_gettext("Event Starts: {start} ({tz}) (Time to be confirmed)").format(
+                    start=formatted_start, tz=tz
+                )
+
+        if start.date() != (end.date() if end else None):
+            if all_day or no_end_time and schedule_type in (ScheduleType.REGULAR, ScheduleType.MULTI_DAY):
+                return lazy_gettext("Event Starts: {start} - Event Ends: {formatted_end} ({tz})").format(
+                    start=formatted_start, end=formatted_end, tz=tz
+                )
+
+        if start.date() == (end.date() if end else None):
+            if no_end_time or all_day and schedule_type in (ScheduleType.ALL_DAY, ScheduleType.NO_DURATION):
+                return lazy_gettext("Event Starts: {start} ({tz})").format(start=formatted_start, tz=tz)
+
+        if no_end_time and start.date():
+            return lazy_gettext("Event Starts: {start} ({tz})").format(start=formatted_start, tz=tz)
+
+        if schedule_type == ScheduleType.REGULAR:
+            return lazy_gettext("Event Starts: {start_time} to {end_time} - On Date: {date} ({tz})").format(
+                start_time=formatted_start.split()[1],
```

Comment:
this assumes specific format, but in different locales time might be before the date. we should use configured time format here

Reply:
I see, but we used `'DATETIME_FORMAT_LONG'` here, which is probably the same for different locales, isn't it?

Reply:
it's different, and there might be other things besides date and time in it like `10:00 le 20.12.2023` for fr

Reply:
I see, can we use these format `TIME_FORMAT_SHORT` for time  and `DATE_FORMAT_SHORT` for date  ?